zboxcore/sdk/allocation.go

```diff
@@ -531,6 +548,90 @@ func (a *Allocation) DownloadFile(localPath string, remotePath string, verifyDow
 	return a.downloadFile(localPath, remotePath, DOWNLOAD_CONTENT_FULL, 1, 0, numBlockDownloads, verifyDownload, status)
 }
 
+func (a *Allocation) DoMultiOperation(operations []OperationRequest) error {
+	if len(operations) == 0 {
+		return nil
+	}
+	if !a.isInitialized() {
+		return notInitialized
+	}
+	var mo MultiOperation
+	mo.allocationObj = a
+	mo.operationMask = zboxutil.NewUint128(1).Lsh(uint64(len(a.Blobbers))).Sub64(1)
+	mo.maskMU = &sync.Mutex{}
+	mo.ctx, mo.ctxCncl = context.WithCancel(a.ctx)
+	mo.Consensus = Consensus{
+		consensusThresh: a.consensusThreshold,
+		fullconsensus:   a.fullconsensus,
+	}
+	mo.connectionID = zboxutil.NewConnectionId()
+	for _, op := range operations {
+		switch op.OperationType {
+
+		case constants.FileOperationRename:
+			renameOp := &RenameOperation{}
+			renameOp.build(op.RemotePath, op.DestName, mo.operationMask, mo.maskMU, mo.consensusThresh, mo.fullconsensus, mo.ctx)
+			err := renameOp.Verify(a)
+			if err != nil {
+				return err
+			}
+			mo.operations = append(mo.operations, renameOp)
```

Comment:
you better make simple constructor for every operation and hide initialization there, so it could be

```go
renameOp:=NewRenameOperation(op.RemotePath, op.DestName, mo.operationMask, mo.maskMU, mo.consensusThresh, mo.fullconsensus, mo.ctx)
```

Reply:
Done